### PR TITLE
New gobal fast property search view.

### PR DESCRIPTION
**Pull request by zanthrash**
Updating tslint to include console.log and console.warn.

@anotherchrisberry @icfantv PTAL

### CODE REVIEW

File: app/scripts/modules/netflix/fastProperties/dataNav/fastProperties.controller.js [hunk 1]
```diff
@@ -1,65 +1,252 @@
...
+        let len = listOfPredicate.length;
+        while(idx < len ) {
+          if(!listOfPredicate[idx].call(this, property)) {
+            return false;
+          }
+          idx += 1;
+        }
+        return true;
+      };
+    };
```
Comment (anotherchrisberry) [inline app/scripts/modules/netflix/fastProperties/dataNav/fastProperties.controller.js]: Can this `while` loop be replaced with an `every` call? i.e.
```javascript
let allPass = (listOfPredicates) => {
  return (property) => listOfPredicates.every(predicate => predicate(property));
};
```

File: app/scripts/modules/netflix/fastProperties/dataNav/fastProperties.controller.js [hunk 2]
```diff
@@ -1,65 +1,252 @@
...
+        vm.searchError = undefined;
+
+        fastPropertyReader.search(vm.searchTerm).then((data) => {
+          vm.searchResults = data.propertiesList.map((fp) => {
+            fp.scope = extractFastPropertyScopeFromId(fp.propertyId);
+            fp.appId = fp.appId || 'All (Global)';
+            return fp;
+          });
+          return vm.searchResults;
+        }).then((searchResutls) => {
```
Comment (anotherchrisberry) [inline app/scripts/modules/netflix/fastProperties/dataNav/fastProperties.controller.js]: s/searchResutls/searchResults

File: app/scripts/modules/netflix/fastProperties/dataNav/fastPropertyFilter.component.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    'createFilterTag': '='
+  };
+  template: string = `<input type="search" class="form-control" placeholder="Filters: type '?'">`;
+  restrict: string = 'E';
+  fields: string[] = ['app', 'env', 'region', 'stack', 'cluster'];
+
+  link(scope: any, el: any) {
+    let input = el.children('input');
+
+    let getScopeAttrabuteList = (scopeName: string) => {
```
Comment (anotherchrisberry) [inline app/scripts/modules/netflix/fastProperties/dataNav/fastPropertyFilter.component.ts]: s/Attrabute/Attribute

File: app/scripts/modules/netflix/fastProperties/dataNav/fastPropertyFilterSearch.component.ts [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+        acc.push({category: scopeKey, results: ['none', scopeValue]});
+      }
+    }
+    return acc;
+  }
+}
+
+
+
+class FastPropertyFilterSearchDirective implements ng.IComponentOptions {
```
Comment (anotherchrisberry) [inline app/scripts/modules/netflix/fastProperties/dataNav/fastPropertyFilterSearch.component.ts]: should `Directive` be `Component` here?

File: app/scripts/modules/netflix/fastProperties/dataNav/fastPropertyFilterSearch.component.ts [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+    }
+  }
+
+  private reset() {
+    this.querying = false;
+    this.query = null;
+    this.showSearchResults = false;
+    this.showAllCategories = false;
+    this.focussedResult = null;
+    // this.$element.find('input').focus();
```
Comment (anotherchrisberry) [inline app/scripts/modules/netflix/fastProperties/dataNav/fastPropertyFilterSearch.component.ts]: remove?

File: app/scripts/modules/netflix/fastProperties/dataNav/fastPropertyFilterSearch.component.ts [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+  public properties: any;
+
+  static get $inject() {
+    return [
+      '$element', '$log'
+    ];
+  }
+
+  constructor(
+    private $element: any,
```
Comment (anotherchrisberry) [inline app/scripts/modules/netflix/fastProperties/dataNav/fastPropertyFilterSearch.component.ts]: the `$element` can be typed to `JQuery` here

File: app/scripts/modules/netflix/fastProperties/dataNav/fastPropertyFilterSearch.component.ts [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+    private $log: any
```
Comment (anotherchrisberry) [inline app/scripts/modules/netflix/fastProperties/dataNav/fastPropertyFilterSearch.component.ts]: ng.ILogService

File: app/scripts/modules/netflix/fastProperties/dataNav/fastPropertyFilterSearch.component.ts [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+
+  public properties: any;
+
+  static get $inject() {
+    return [
+      '$element', '$log'
+    ];
+  }
+
+  constructor(
```
Comment (anotherchrisberry) [inline app/scripts/modules/netflix/fastProperties/dataNav/fastPropertyFilterSearch.component.ts]: prefer making this `public` (per @icfantv , this is required for Angular 2)

File: app/scripts/modules/netflix/fastProperties/dataNav/fastPropertyFilterSearch.component.ts [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+    private $element: any,
+    private $log: any
+  ) {}
+
+  public $onInit(): void {
+    this.createFilterCategories(this.properties);
+  }
+
+
+  private displayAllCategories() {
```
Comment (anotherchrisberry) [inline app/scripts/modules/netflix/fastProperties/dataNav/fastPropertyFilterSearch.component.ts]: return type (`void`)

File: app/scripts/modules/netflix/fastProperties/dataNav/fastPropertyFilterSearch.component.ts [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+    this.filteredCategories = this.categories;
+    this.showSearchResults = true;
+  }
+
+  public displayResults() {
```
Comment (anotherchrisberry) [inline app/scripts/modules/netflix/fastProperties/dataNav/fastPropertyFilterSearch.component.ts]: return type (`void`)

File: app/scripts/modules/netflix/fastProperties/dataNav/fastPropertyFilterSearch.component.ts [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+    if (this.query) {
+      this.executeQuery();
+    } else {
+      this.displayAllCategories();
+    }
+  }
+
+  private reset() {
```
Comment (anotherchrisberry) [inline app/scripts/modules/netflix/fastProperties/dataNav/fastPropertyFilterSearch.component.ts]: return type (`void`)

File: app/scripts/modules/netflix/fastProperties/dataNav/main.html [hunk 12]
```diff
@@ -2,29 +2,40 @@
...
+             <!--ng-class="{active: $state.includes('**.rollouts.**')}">-->
+            <!--Property Rollouts-->
+          <!--</a>-->
+        <!--</li>-->
+        <!--<li>-->
+          <!--<a ui-sref=".diffs" disabled-->
+             <!--ng-class="{active: $state.includes('**.diffs.**')}">-->
+            <!--Diff Utility-->
+          <!--</a>-->
+        <!--</li>-->
```
Comment (anotherchrisberry) [inline app/scripts/modules/netflix/fastProperties/dataNav/main.html]: remove

File: app/scripts/modules/netflix/fastProperties/dataNav/properties.html [hunk 13]
```diff
@@ -1,57 +1,84 @@
...
-               ng-change="fp.filter(fp.applications)"/>
+      <form class="form form-inline">
+        <div class="form-group">
+          <label for="propertySearch">Search</label>
+          <input type="search"
+                 id="propertySearch"
+                 placeholder="Search properties"
+                 class="form-control"
+                 style="min-width: 220px; width: auto;"
+
```
Comment (anotherchrisberry) [inline app/scripts/modules/netflix/fastProperties/dataNav/properties.html]: remove

File: app/scripts/modules/netflix/fastProperties/dataNav/properties.html [hunk 14]
```diff
@@ -1,57 +1,84 @@
...
+                    ng-click="fp.setGroupTo(fp.groupNames.APP)">App
+            </button>
+            <button class="btn btn-default" ng-class="{active: fp.selectedGroupIs(fp.groupNames.PROPERTY)}"
+                    ng-click="fp.setGroupTo(fp.groupNames.PROPERTY)">Property
+            </button>
+          </div>
+        </div>
+
+        <div class="form-group" ng-if="fp.propertiesList">
+          <!--<fast-property-filter properties="fp.searchResults" filters="filters" create-filter-tag="createFilterTag"></fast-property-filter>-->
```
Comment (anotherchrisberry) [inline app/scripts/modules/netflix/fastProperties/dataNav/properties.html]: remove?

File: app/scripts/modules/netflix/fastProperties/dataNav/fastPropertyFilterSearch.component.ts [hunk 15]
```diff
@@ -0,0 +1,215 @@
...
+    'createFilterTag': '='
+  };
+  public controller: ng.IComponentController = FastPropertyFilterSearchController;
+  public controllerAs: string = 'fpFilter';
+  public templateUrl: string = require('./fastPropertyFilterSearch.component.html');
+
+}
+
+
+const moduleName = 'spinnaker.netflix.fastPropertyFilterSearch.directive';
```
Comment (anotherchrisberry) [inline app/scripts/modules/netflix/fastProperties/dataNav/fastPropertyFilterSearch.component.ts]: prefer exporting this, e.g.
```javascript
export const FAST_PROPERTY_SEARCH_COMPONENT = 'spinnaker.netflix.fastPropertyFilterSearch.component';
```
then doing an explicit import on the field where required

File: app/scripts/modules/netflix/fastProperties/globalFastPropertyDetails.controller.js [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+                return propertyWithScope;
+              });
+          },
+        }
+
+      }).result.then(refreshApp);
+    };
+
+
+    let refreshApp = () => angular.noop;
```
Comment (anotherchrisberry) [inline app/scripts/modules/netflix/fastProperties/globalFastPropertyDetails.controller.js]: what is this doing?

Comment (zanthrash) [inline app/scripts/modules/netflix/fastProperties/globalFastPropertyDetails.controller.js]: noop

**Pull request merged**